qiskit/transpiler/passes/utils/gate_direction.py

```diff
@@ -112,15 +116,16 @@ def run(self, dag):
             )
 
         trivial_layout = Layout.generate_trivial_layout(*dag.qregs.values())
+        layout_map = trivial_layout.get_virtual_bits()
 
         for node in dag.two_qubit_ops():
             control = node.qargs[0]
             target = node.qargs[1]
 
-            physical_q0 = trivial_layout[control]
-            physical_q1 = trivial_layout[target]
+            physical_q0 = layout_map[control]
+            physical_q1 = layout_map[target]
 
-            if self.coupling_map.distance(physical_q0, physical_q1) != 1:
+            if self.coupling_map._dist_matrix[physical_q0, physical_q1] != 1:
```

Comment:
Same as elsewhere, can we generate the distance matrix once with `self.coupling_map.distance_matrix` and access from there?

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/657d605285fa19576d1aeb865dd57b8e9a99f6bc